jsx/Form.js

```diff
@@ -134,6 +134,131 @@ var FormElement = React.createClass({
   }
 });
 
+/**
+ * Datalist Component
+ * React wrapper for a searchable dropdown
+ */
+var DatalistElement = React.createClass({
+
+  propTypes: {
+    name: React.PropTypes.string.isRequired,
+    options: React.PropTypes.object.isRequired,
+    id: React.PropTypes.string,
+    strictDatalist: React.PropTypes.bool,
+    label: React.PropTypes.string,
+    value: React.PropTypes.oneOfType([
+      React.PropTypes.string,
+      React.PropTypes.array
+    ]),
+    class: React.PropTypes.string,
+    disabled: React.PropTypes.bool,
+    required: React.PropTypes.bool,
+    hasError: React.PropTypes.bool,
+    errorMessage: React.PropTypes.string,
+    onUserInput: React.PropTypes.func
+  },
+
+  getDefaultProps: function() {
+    return {
+      name: '',
+      options: {},
+      strictDatalist: false,
```

Comment:
Perhaps `strictDatalist` should default to true. Since `options` is required, it seems to me that this will be the more likely use-case for the searchable dropdown.